lib/GitterUser.js

```diff
@@ -0,0 +1,128 @@
+"use strict";
+
+var Promise = require("bluebird");
+
+var rp = require("request-promise");
+
+// miliseconds of interval between presence=online updates
+var ONLINE_PERIOD = 4 * 60 * 1000;
+
+// miliseconds of grace period before users count as offline
+var OFFLINE_GRACE = 3 * 60 * 1000;
+
+function GitterUser(bridge, remoteUserModel) {
+  this._bridge = bridge;
+  this._remoteUserModel = remoteUserModel;
+
+  this._presenceRooms = {};
+  this._ispresent = false;
+  this._presenceinterval = null;
+  this._presencedelay = null;
```

Comment:
After reading most of the file, I now know that these are actually timer IDs for use with `setTimeout` and `setInterval`. I would `s/_presencedelay/_presenceDelayId/` and `s/_presenceinterval/_presenceIntervalId/` to make it clearer what these are meant to be (IDs for use when cancelling timers) rather than letting me speculate any further (a `Number` for ms of delay? A function which will be called when the delay timer expires?).